src/table/table-role/__tests__/grid-navigation-context.test.tsx

```diff
@@ -0,0 +1,72 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import React, { useCallback, useRef } from 'react';
+import { render } from '@testing-library/react';
+import { GridNavigationContext } from '../../../../lib/components/table/table-role/grid-navigation-context';
+import { useGridNavigationFocusable } from '../../../../lib/components/table/table-role';
+import { renderWithGridNavigation } from './utils';
+
+function Button(props: React.HTMLAttributes<HTMLButtonElement>) {
+  const buttonRef = useRef<HTMLButtonElement>(null);
+  const { tabIndex } = useGridNavigationFocusable(buttonRef, { tabIndex: props.tabIndex });
+  return <button {...props} ref={buttonRef} tabIndex={tabIndex} />;
+}
+
+function ButtonAlt(props: React.HTMLAttributes<HTMLButtonElement>) {
```

Comment:
For better understanding, should we call these buttons something like `ButtonWithRef` and `ButtonWithCallback`?